lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -190,8 +190,8 @@ GetTypeAlias(swift::Demangle::Demangler &dem,
 }
 
 /// Find a Clang type by name in the modules in \p module_holder.
-TypeSP TypeSystemSwiftTypeRef::LookupClangType(StringRef name) {
-  auto lookup = [](Module &M, StringRef name) -> TypeSP {
+TypeSP TypeSystemSwiftTypeRef::LookupClangType(ConstString name) {
```

Comment:
Why did this change to `ConstString`?

Reply:
To elaborate, it looks as though the use of `ConstString` could be internal to the function rather.

Reply:
Because the cache is using `char*` as keys, and FindTypes takes ConstStrings anyway.

Reply:
My thinking was that by exposing the fact that this will end up as a ConstString in any case, callers can make a decision to convert to ConstString earlier which could lead to more efficient code. In practice the name is always a demangle text node, so this argument doesn't really work. Should I change it back to StringRef?

Reply:
The explanation works for me. I don't have any instinct on the theory vs practice, your call.